.github/actions/e2e_benchmark/evaluate/compare.py

```diff
@@ -133,20 +157,28 @@ def compare_test(test, subtest, metric, bench_prev, bench_curr) -> str:
             if is_bigger_better(bench_suite=metric):
                 ratio_num = val_curr / val_prev
                 if ratio_num < ALLOWED_RATIO_DELTA.get(metric, 1):
-                    set_failed()
+                    self.set_failed()
             else:
                 ratio_num = val_prev / val_curr
                 if ratio_num > ALLOWED_RATIO_DELTA.get(metric, 1):
-                    set_failed()
-                
+                    self.set_failed()
+
             ratio_num = round(ratio_num, 3)
             emoji = PROGRESS[int(ratio_num >= 1)]
             ratio = f'{ratio_num} {emoji}'
 
         return f'| {subtest} | {metric} ({unit}) | {val_curr} | {val_prev} | {ratio} |'
 
-def set_failed() -> None:
-    os.environ['COMPARISON_SUCCESS'] = str(False)
+    def set_failed(self) -> None:
+        os.environ['COMPARISON_SUCCESS'] = str(False)
```

Comment:
This function can be moved outside the class. It does not use `self` in any form and has global state due to the OS environment.